src/Components/OrderRow.ts

```diff
@@ -0,0 +1,112 @@
+import Component, { Template } from '@/core/Component';
+import { ORDER_STORE } from '@/Stores/orderStore/constants';
+import { dispatch } from '@/Stores/orderStore/orderStore';
+import Order from '@/Model/Order';
+
+class OrderRow extends Component {
+  $container!: HTMLElement;
+  $optionGroups!: HTMLElement[];
+  order!: Order;
+
+  protected componentDidMonted: () => void = () => {
+    this.addEventListener(ORDER_STORE.event, e => {
+      const { type, payload } = e.detail
+      if (type === ORDER_STORE.types.UPDATE && this.order.id === payload.id) {
+        this.setOrder(payload as Order);
+      }
+    });
+  };
+
+  public setOrder(order: Order) {
+    this.order = order;
+    this.render();
+  }
+
+  private createCell(title: string, textContent?: string | number) {
+    const cell = document.createElement('div');
+    cell.classList.add('cell');
+    cell.setAttribute('data-title', title);
+    cell.textContent = String(textContent || '');
+    return cell;
+  }
+
+  private appendButtonIcon(cell: HTMLDivElement) {
+    const span = document.createElement('span');
+    span.classList.add('edit-order');
+
+    const cellType = cell.getAttribute('data-title')!;
+    const { icon, onClick } = this.getButtonAttribute(cellType);
+
+    const i = document.createElement('i');
+    i.classList.add('fa-solid');
+    i.classList.add(icon);
+    span.appendChild(i);
+    span.addEventListener('click', onClick);
+    cell.appendChild(span);
+    return cell;
+  }
+
+  private getButtonAttribute = (type: string) => {
+    if (type === '수정하기') {
+      return {
+        icon: 'fa-pen',
+        onClick: () => {
+          this.$optionGroups.forEach($optionGroup => {
+            const CONTENT_EDITABLE = 'contenteditable';
+            const isEditable = $optionGroup.getAttribute(CONTENT_EDITABLE);
+            if (isEditable) {
+              return $optionGroup.removeAttribute(CONTENT_EDITABLE);
+            }
+            $optionGroup.setAttribute(CONTENT_EDITABLE, 'true');
+          });
+        },
+      };
+    }
+
+    return {
+      icon: 'fa-trash-can',
+      onClick: () => {
+        dispatch({ type: ORDER_STORE.types.DELETE, payload: this.order });
+        this.destroy();
+      },
+    };
+  };
+
+  private destroy = () => {
+    this.$container.remove();
+    this.remove();
+  };
+
+  protected template: () => Template = () => {
+    const tableRowWrapper = document.createElement('div');
+    tableRowWrapper.classList.add('table-row');
+    this.$container = tableRowWrapper;
+
+    if (!this.order) {
+      const div = document.createElement('div');
+      div.textContent = 'loading...';
+      return {
+        parent: tableRowWrapper,
+        children: [div],
+      };
+    }
+
+    const idCell = this.createCell('id', 1);
+    const cafe = this.createCell('메뉴명', this.order.drink.menuName);
```

Comment:
```suggestion
    const cafe = this.createCell('메뉴명', this.order.menuName());
```
이렇게 drink를 거치지 않고 menuName을 가저 올수 있도록 캡슐화 해보는건 어떨까요~? 😀

Reply:
오! 감사합니다!!

아직 객체지향에 익숙치 않은게 살살 티가 나네요 🥲